sequencers/modeshape-sequencer-epub/src/main/java/org/modeshape/sequencer/epub/EpubMetadataSequencer.java

```diff
@@ -0,0 +1,198 @@
+            setPropertyIfMetadataPresent(propertyNode, EpubMetadataLexicon.GROUP_POSITION, value.getGroupPosition());
+            setPropertyIfMetadataPresent(propertyNode, EpubMetadataLexicon.IDENTIFIER_TYPE, value.getIdentifierType());
+            setPropertyIfMetadataPresent(propertyNode, EpubMetadataLexicon.SCHEME, value.getScheme());
+
+            if (value.getAlternateScript() != null) {
+                Node alternateScriptNode = propertyNode.addNode(EpubMetadataLexicon.ALTERNATE_SCRIPT_NODE, EpubMetadataLexicon.ALTERNATE_SCRIPT_NODE);
+                setPropertyIfMetadataPresent(alternateScriptNode, EpubMetadataLexicon.VALUE, value.getAlternateScript().getValue());
+                setPropertyIfMetadataPresent(alternateScriptNode, EpubMetadataLexicon.LANGUAGE_CODE, value.getAlternateScript().getLanguage());
+            }
+        }
+    }
+
+    private void setPropertyIfMetadataPresent( Node node,
+                                               String propertyName,
+                                               Object value ) throws RepositoryException {
+        if (value != null) {
+            if (value instanceof String && !StringUtil.isBlank((String) value)) {
+                node.setProperty(propertyName, (String) value);
+            } else if (value instanceof Boolean) {
+                node.setProperty(propertyName, (Boolean) value);
+            } else if (value instanceof Long) {
+                node.setProperty(propertyName, (Long) value);
+            } else if (value instanceof Integer) {
+                node.setProperty(propertyName, new Long((Integer) value));
+            } else if (value instanceof Double) {
+                node.setProperty(propertyName, (Double) value);
+            } else if (value instanceof Calendar) {
+                node.setProperty(propertyName, (Calendar) value); 
+            } else if (value instanceof byte[]) {
+                InputStream is = new ByteArrayInputStream((byte []) value);
+                javax.jcr.Binary binaryProperty = node.getSession().getValueFactory().createBinary(is);
+                node.setProperty(propertyName, binaryProperty);
+            } else if (value instanceof List<?>) {
+                ValueFactory vf = node.getSession().getValueFactory();
+                List<Value> values = new ArrayList<Value>();
+                for (Object val : (List<?>) value) {
+                    if (val instanceof String) {
+                        values.add(vf.createValue((String) val));
+                    }
+                }
+                if (values.size() > 0) {
+                    Value[] valuesArray = new Value[values.size()];
+                    values.toArray(valuesArray);
+                    node.setProperty(propertyName, valuesArray);
+                }
+            } else {
+                throw new IllegalArgumentException(String.format("The value of the property %s has unknown type and couldn't be saved.", propertyName));
```

Comment:
do we really want to throw an exception here ? Wouldn't perhaps logging a warning and continuing be more appropriate ?